priv/scim.abnf

```diff
@@ -2,27 +2,29 @@
 ; https://www.rfc-editor.org/rfc/rfc7644#section-4
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-FILTER    = attrExp / logExp / valuePath / *1"not" "(" FILTER ")"
+FILTER    = (attrExp / valuePath / [NOT SP] "(" FILTER ")") [SP AND-OR SP FILTER]
 
 valuePath = attrPath "[" valFilter "]"
             ; FILTER uses sub-attributes of a parent attrPath
 
-valFilter = attrExp / logExp / *1"not" "(" valFilter ")"
+valFilter = (attrExp / [NOT SP] "(" valFilter ")") [SP AND-OR SP valFilter]
 
 attrExp   = (attrPath SP "pr") /
             (attrPath SP compareOp SP compValue)
 
-logExp    = FILTER SP ("and" / "or") SP FILTER
-
 compValue = false / null / true / number / string
 	  ; rules from JSON (RFC 7159)
 
+NOT = "not"
+
+AND-OR = "and" / "or"
+
 compareOp = "eq" / "ne" / "co" /
 		 "sw" / "ew" /
 		 "gt" / "lt" /
 		 "ge" / "le"
 
-attrPath  = [URI ":"] ATTRNAME *1subAttr
+attrPath  = (URI / ATTRNAME) *1subAttr
```

Comment:
This is a hack because

1. URI is super flexible, `a:::::` is a valid URI
2. generated parser is greedy and does not perform backtracking

`uri:a:b:c` will be completely matched by `URI` and there will be nothing left for `ATTRNAME`, hence failing the `attrPath`

Reply:
You will have to match on `{:uri, ...}` and get the last part of the uri, or `{:attrname, ...}`.

Beware that URI is super set of ATTRNAME, the last part may be invalid attrname.